spec/unit/recipe_spec.rb

```diff
@@ -161,7 +161,76 @@
         zm_resource # force let binding evaluation
         expect { run_context.resource_collection.resources(:zen_master => "klopp") }.to raise_error(Chef::Exceptions::ResourceNotFound)
       end
+    end
+
+    describe "when cloning resources" do
+      it "should emit a 3694 warning when attributes change" do
+        recipe.zen_master "klopp" do
+          something "bvb"
+        end
+        expect(Chef::Log).to receive(:warn).at_least(:once)
```

Comment:
Can we assert that this is at least the `CHEF-3694` warning (either initialize the logger with a `StringIO` and check content, or tighten up the expectation with a string or regex)? Otherwise it's possible we do something else that triggers a warning but don't see it in the tests because of this loose expectation.